lib/iex/lib/iex/helpers.ex

```diff
@@ -1068,6 +1068,14 @@ defmodule IEx.Helpers do
     path = Path.expand(path)
 
     if not optional? or File.exists?(path) do
+      imported_paths = Process.get({__MODULE__, :imported_paths}, MapSet.new())
```

Comment:
I don't think we need to detect loops themselves, only if we are expanding to yourself. This could be done by passing `__CALLER__` to this function and checking if `Path.expand(path) == __CALLER__.file`. This also guarantees we won't pollute the process dictionary (which we don't clean right now and I don't think we could). :)

Reply:
I guess with your approach we would still have the issue that the test shows, but I agree that's probably way less common than the issue that user reported (that the path resolves to itself). Will work on a patch!

Reply:
@josevalim I tried that but the `caller.file` is always `iex`. Is there a way to update calling environment?

Here's a diff against master of what I tried:

```diff
diff --git a/lib/iex/lib/iex/helpers.ex b/lib/iex/lib/iex/helpers.ex
index 773ea0376..dbb66aeca 100644
--- a/lib/iex/lib/iex/helpers.ex
+++ b/lib/iex/lib/iex/helpers.ex
@@ -1057,18 +1057,20 @@ defmodule IEx.Helpers do
   in behalf of the user.
   """
   defmacro import_file_if_available(path) when is_binary(path) do
-    import_file_if_available(path, true)
+    import_file_if_available(path, __CALLER__, true)
   end
 
   defmacro import_file_if_available(_) do
     raise ArgumentError, "import_file_if_available/1 expects a literal binary as its argument"
   end
 
-  defp import_file_if_available(path, optional?) when is_binary(path) do
-    path = Path.expand(path)
+  defp import_file_if_available(path, caller, optional?) when is_binary(path) do
+    if Path.expand(path) == caller.file do
+      raise ArgumentError, "attempting to import #{inspect(path)} from itself"
+    end
 
     if not optional? or File.exists?(path) do
-      path |> File.read!() |> Code.string_to_quoted!(file: path)
+      path |> File.read!() |> Code.string_to_quoted!(file: path)
     end
   end
 
@@ -1100,7 +1102,7 @@ defmodule IEx.Helpers do
   """
   @doc since: "1.4.0"
   defmacro import_file(path) when is_binary(path) do
-    import_file_if_available(path, false)
+    import_file_if_available(path, __CALLER__, false)
   end
 
   defmacro import_file(_) do
@@ -1110,7 +1112,7 @@ defmodule IEx.Helpers do
   @doc false
   @deprecated "Use import_file_if_available/1 instead"
   defmacro import_file(path, opts) when is_binary(path) and is_list(opts) do
-    import_file_if_available(path, Keyword.get(opts, :optional, false))
+    import_file_if_available(path, __CALLER__, Keyword.get(opts, :optional, false))
   end
 
   @doc """
```